Lib/test/test_calendar.py

```diff
@@ -735,12 +735,15 @@ def get_weekday_names(width):
     def test_locale_calendar_long_weekday_names(self):
         names = (datetime.date(2001, 1, i+1).strftime('%A') for i in range(7))
         max_length = max(map(len, names))
-        if max_length <= 9:
-            self.skipTest('weekday names are too short')
+        abbrev_names = (datetime.date(2001, 1, i+1).strftime('%a') for i in range(7))
+        abbrev_max_length = max(map(len, abbrev_names))
+
+        if max_length - abbrev_max_length < 3:
+            self.skipTest('standard and abbreviated weekday names have too similar lengths')
 
         def get_weekday_names(width):
             return calendar.TextCalendar().formatweekheader(width).split()
-        self.assertEqual(get_weekday_names(4), get_weekday_names(9))
+        self.assertEqual(get_weekday_names(abbrev_max_length+1), get_weekday_names(max_length-1))
```

Comment:
```suggestion
        self.assertEqual(get_weekday_names(abbrev_max_length), get_weekday_names(max_length-1))
```
We could also use a weaker test for widths 8 and 9, this was enough to test the original issue, but a stricter test can catch more unintentional changes in future.

Reply:
Ah yes, that makes sense.